### DISCUSSION TITLE
Can this be improved in the clap headers?

### DISCUSSION THREAD

**Discussion by logos-maker** [free-audio/clap#144] [Ideas]
In the declaration of the get_info function in clap/ext/params.h

Can param_index be a const in the function  declaration?

Like...
bool (*get_info)(const clap_plugin_t *plugin, const uint32_t   param_index, clap_param_info_t   *param_info);

So newbie plugin developers get a compilation error if they mess with it, as it do nothing anyways.

**Comment by robbert-vdh** [comment]
Const-value types should generally not be in header function signatures as they're purely a detail of the implementation and don't convey any new information about the API.
